### ISSUE TITLE
Exporting a member of a class under the same name throws: "Cannot read property 'contains' of undefined"

### ISSUE THREAD

**Issue by niieani** [decaffeinate/decaffeinate#335]
This breaks: 

```coffee
class ArrayResult
    property: 'a'

module.exports.property = ArrayResult::property
```
[(repl)](http://decaffeinate-project.org/repl/#?evaluate=false&code=class%20ArrayResult%0A%20%20%20%20property%3A%20'a'%0A%0Amodule.exports.property%20%3D%20ArrayResult%3A%3Aproperty)

----

It seems only to break when the exported property has the same name as the property name of the class. When we change 'property' (in either of those) to something else, it magically starts to work (changed `module.exports.property` to `module.exports.otherProperty`:

```coffee
class ArrayResult
    property: 'a'

module.exports.otherProperty = ArrayResult::property
```
[(repl)](http://decaffeinate-project.org/repl/#?evaluate=false&code=class%20ArrayResult%0A%20%20%20%20property%3A%20'a'%0A%0Amodule.exports.otherProperty%20%3D%20ArrayResult%3A%3Aproperty)

----

Expected result:

```js
class ArrayResult {
    property = 'a';
}

export let property = ArrayResult.prototype.property;
```

----

I get this error:

```
Cannot read property 'contains' of undefined
```

**Comment by eventualbuddha** [comment]
This is due to a bad design choice in esnext that will be fixed by https://github.com/esnext/esnext/issues/101. In particular, this error is caused by this line:

```js
module.magicString.insertLeft(leftRightOfAssignment(firstElement).left.start, '{ ');
```

Since the result of `leftRightOfAssignment(firstElement).left` is an `Identifier` node that was generated by the `modules.commonjs` plugin, it doesn't have location information (i.e. no `start` property). Therefore magic-string fails when we pass it an `undefined` insertion position.

**Comment by eventualbuddha** [comment]
This is now fixed in v2.19.5 and converts to this:

```js
class ArrayResult {
    property = 'a';
}

export let { property } = ArrayResult.prototype;
```

**Comment by niieani** [comment]
Fantastic, thank you Brian.
